lib/elixir/lib/kernel/typespec.ex

```diff
@@ -841,7 +841,7 @@ defmodule Kernel.Typespec do
     {{:type, line(meta), :fun, args}, state}
   end
 
-  defp typespec({name, meta, args}, vars, caller, state) do
+  defp typespec({name, meta, args}, vars, caller, state) when is_atom(name) do
```

Comment:
Remote calls are already handled above, so I think it should never be anything else than an atom here?